planner_cspace/src/distance_map.cpp

```diff
@@ -0,0 +1,330 @@
+  {
+    for (d[1] = -range_rough; d[1] <= range_rough; d[1]++)
+    {
+      if (d[0] == 0 && d[1] == 0)
+        continue;
+      if (d.sqlen() > range_rough * range_rough)
+        continue;
+
+      SearchDiffs diffs;
+
+      const float grid_to_len = d.gridToLenFactor();
+      const int dist = d.len();
+      const float dpx = static_cast<float>(d[0]) / dist;
+      const float dpy = static_cast<float>(d[1]) / dist;
+      Astar::Vecf pos(0, 0, 0);
+      for (int i = 0; i < dist; i++)
+      {
+        Astar::Vec ipos(pos);
+        if (diffs.pos.size() == 0 || diffs.pos.back() != ipos)
+        {
+          diffs.pos.push_back(std::move(ipos));
+        }
+        pos[0] += dpx;
+        pos[1] += dpy;
+      }
+      diffs.grid_to_len = grid_to_len;
+      diffs.d = d;
+      diffs.euclid_cost = model->euclidCostRough(d);
+      search_diffs_.push_back(std::move(diffs));
+    }
+  }
+}
+
+void DistanceMap::update(
+    const Astar::Vec& s, Astar::Vec e,
+    const Rect& rect)
+{
+  e[2] = 0;
+
+  Astar::Vec p_cost_min;
+  float cost_min = std::numeric_limits<float>::max();
+  for (Astar::Vec p(0, rect.min[1], 0); p[1] < rect.max[1]; p[1]++)
+  {
+    for (p[0] = rect.min[0]; p[0] < rect.max[0]; p[0]++)
+    {
+      if (cost_min > g_[p])
+      {
+        p_cost_min = p;
+        cost_min = g_[p];
+      }
+    }
+  }
+  if (cost_min == 0.0)
+  {
+    // Gorl is in the updated area
+    create(s, e);
+    return;
+  }
+  pq_open_.clear();
+  pq_erase_.clear();
+
+  if (cost_min != std::numeric_limits<float>::max())
+    pq_erase_.emplace(cost_min, cost_min, p_cost_min);
+
+  while (true)
+  {
+    if (pq_erase_.size() < 1)
+      break;
+    const Astar::PriorityVec center(pq_erase_.top());
+    const Astar::Vec p = center.v_;
+    pq_erase_.pop();
+
+    if (g_[p] == std::numeric_limits<float>::max())
+      continue;
+    g_[p] = std::numeric_limits<float>::max();
+
+    for (Astar::Vec d(-1, 0, 0); d[0] <= 1; d[0]++)
+    {
+      for (d[1] = -1; d[1] <= 1; d[1]++)
+      {
+        if (!((d[0] == 0) ^ (d[1] == 0)))
+          continue;
+        Astar::Vec next = p + d;
+        next[2] = 0;
+        if ((unsigned int)next[0] >= (unsigned int)p_.size[0] ||
+            (unsigned int)next[1] >= (unsigned int)p_.size[1])
+          continue;
+        const float gn = g_[next];
+        if (gn == std::numeric_limits<float>::max())
+          continue;
+        if (gn < cost_min)
+        {
+          pq_open_.emplace(gn, gn, next);
+          continue;
+        }
+        pq_erase_.emplace(gn, gn, next);
+      }
+    }
+  }
+  if (pq_open_.size() == 0)
+  {
+    pq_open_.emplace(-p_.euclid_cost[0] * 0.5, -p_.euclid_cost[0] * 0.5, e);
+  }
+
+  for (Astar::Vec p(0, 0, 0); p[0] < p_.size[0]; p[0]++)
+  {
+    for (p[1] = 0; p[1] < p_.size[1]; p[1]++)
+    {
+      const float gp = g_[p];
+      if ((gp > rough_cost_max_) && (gp != std::numeric_limits<float>::max()))
+      {
+        pq_open_.emplace(gp, gp, p);
+      }
+    }
+  }
```

Comment:
L309 seems not to be covered by the test. 
It would be better to add another test to show the necessity of this loop,

Reply:
This block was added by https://github.com/at-wat/neonavigation/commit/cd3cff4ae968dc26763523b4d400462a1d620dbb
and no longer had meaning after https://github.com/at-wat/neonavigation/pull/518/files